java/java.hints.test/src/org/netbeans/modules/java/hints/test/api/HintTest.java

```diff
@@ -381,12 +378,27 @@ private void ensureCompilable(FileObject file) throws IOException, AssertionErro
         }
     }
 
+    /**Sets a source level for all Java files used in this test.
+     *
+     * @param sourceLevel the source level to use while parsing Java files
+     * @return itself
+     */
+    public HintTest sourceLevel(int sourceLevel) {
+        assertTrue(sourceLevel >= 8);
+        return this.sourceLevel(Integer.toString(sourceLevel));
+    }
+
     /**Sets a source level for all Java files used in this test.
      *
      * @param sourceLevel the source level to use while parsing Java files
      * @return itself
      */
     public HintTest sourceLevel(String sourceLevel) {
+        try {
+            int valid = sourceLevel.startsWith("1.") ? Integer.parseInt(sourceLevel.substring(2)) : Integer.parseInt(sourceLevel);
+        } catch (NumberFormatException ex) {
+            throw new IllegalArgumentException(ex);
+        }
         this.sourceLevel = sourceLevel;
         return this;
     }
```

Comment:
some tests used `SourceVersion.latest().name()` as source level which would be the full enum name. This would be quietly ignored and cause weird issues.

The expected format is now enforced and the tests were fixed which used it wrong.